src/map/src/cmapclient.cpp

```diff
@@ -191,6 +194,33 @@ bool CMapClient::ChatReply(std::unique_ptr<RoseCommon::CliChat> P) {
   return true;
 }
 
+bool CMapClient::ChatWhisper(std::unique_ptr<RoseCommon::CliWhisper> P) {
+  logger_->trace("CMapClient::ChatWhisper()");
+
+  if (login_state_ != eSTATE::LOGGEDIN) {
+    logger_->warn("Client {} is attempting to whisper before logging in.",
+                  GetId());
+    return true;
+  }
+
+  uint16_t _charID = GetId();
+  std::string _sender_name = entity_.component<BasicInfo>()->name_; 
+	std::string _recipient_name = P->targetId();
+	_recipient_name.erase(_recipient_name.size()-1); // for some reason this string has an extra char
+	std::string _message = P->message();
+	CMapClient *_recipient_client = NULL;
+	entitySystem_->process<BasicInfo, SocketConnector>([&_recipient_client, _recipient_name] (Entity entity) {
+		if(_recipient_name == entity.component<BasicInfo>()->name_)
+			_recipient_client = entity.component<SocketConnector>()->client_;
```

Comment:
if you return false in the if and true outside of the if, it'll magically skip all the entities once it has found the correct one ;)